### PR TITLE
Update master docs content to conform to new docs.crossplane.io

### PR DESCRIPTION
**Pull request by plumbis**
### Description of your changes
This PR updates the docs content to conform with changes required for the new docs.crossplane site deployment. 
This includes changes to front matter and tab generation. 

Functional changes include:
* replacing markdown images with image shortcode to convert images to webp and lazy load at build time.
* `docs/faqs/related_projects.md` - Fixes broken k8s link
* `Makefile` -  moves docs into `/content` instead of `/content/docs` on the website
* `docs/contributing/release-process.md` - Removes reference to "complete"
* `docs/getting-started/create-configuration.md` removes incomplete Alibaba configuration
* `docs/api-docs/_index.md` - removes references to crds.dev
* Removes links that redirected to crds.dev
* `docs/concepts/providers.md` - removes outdated provider table
* Updates multiple packages to pull from Upbound marketplace
* `docs/contributing/docs.md` massive rewrite and expansion to cover all new features and functionality of the updated docs site

**Do not merge this PR until /after/ the new docs site has been deployed.**

### CODE REVIEW

**Comment by plumbis** [comment]
Docs Validation fails because make is rendering the old site that doesn't include one or more shortcodes that are in the new site.

**Review by hasheddan — COMMENTED** [review]
@plumbis you should be able to adjust that here: https://github.com/crossplane/crossplane/pull/3486/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R72

**Comment by plumbis** [comment]
@hasheddan the repo isn't changing. Because the new site features aren't merged yet the test fails. We don't want to update the Makefile since what's there now will be accurate _after_ the site updates. 

If you want to test manually you can pull the [`theme-update` branch](https://github.com/crossplane/crossplane.github.io/tree/theme-update).

File: docs/concepts/managed-resources.md
Comment (hasheddan) [inline docs/concepts/managed-resources.md:35]: ah so nice to get rid of this!

File: docs/contributing/docs.md
Comment (hasheddan) [inline docs/contributing/docs.md:55]: This appears to be repeated above.

File: docs/contributing/docs.md
Comment (hasheddan) [inline docs/contributing/docs.md:60]: Why do we recommend this? You can run the docs site from this repo as outlined in https://github.com/crossplane/crossplane/issues/3461#issuecomment-1324536727. I believe the justification there was that folks may not have `make` installed, but it is going to be installed by default in many distributions. That means they only have to run one command, which feels much simpler than jumping between multiple repositories.

File: docs/contributing/docs.md
Comment (plumbis) [reply docs/contributing/docs.md:55]: This is intentional to try and avoid a mix up between the two repos.

File: docs/contributing/docs.md
Comment (plumbis) [reply docs/contributing/docs.md:60]: >default in many distributions.

But not on mac or Windows.

Comment (hasheddan) [reply docs/contributing/docs.md:60]: They are eventually having to come and interact with this repo, where we ask folks to run `make` commands prior to submitting a PR. I am all for reducing any barrier to entry, but I don't think that this is actually accomplishing that goal as it requires more steps rather than less.

Comment (plumbis) [reply docs/contributing/docs.md]: Suggestion removed.

File: docs/contributing/docs.md
Comment (hasheddan) [reply docs/contributing/docs.md:55]: ah i see -- this is also slightly different from what is said above 👍🏻

File: docs/contributing/docs.md
Comment (hasheddan) [reply docs/contributing/docs.md]: Thanks. I feel like the current status is still pointing folks towards using the Crossplane website repo, and is not suggesting using any of the machinery that is put in place there. We have single commands that ensure you have the correct version of hugo installed, etc. and allow for very quick iteration. Is the reason for not recommending any of this in order to avoid using `make`? If I was helping someone contribute I would say:
1. Clone the Crossplane repo
2. Make changes
3. Run `make docs.run`

This will ensure you have the latest version of the docs site and are using the correct version of hugo. Is there anything we could change about this machinery to make you more inclined to suggest it?

Comment (plumbis) [reply docs/contributing/docs.md]: You're a beat ahead of me. I was just walking through this workflow to add instructions but ran into https://github.com/crossplane/crossplane/issues/3487.

If this is resolved I'll put in detailed steps on the make process.

(also our make configuration [doesn't support Windows](https://github.com/upbound/build/blob/master/makelib/common.mk#L99))

Comment (hasheddan) [reply docs/contributing/docs.md]: @plumbis opened https://github.com/crossplane/crossplane/pull/3488 to fix -- you can consider be signed up to fix any issues with improving the docs workflow in this repo since I am pushing so hard for using it :)

File: docs/faqs/related_projects.md
Comment (negz) [inline docs/faqs/related_projects.md:78]: FWIW we could completely remove this - I haven't heard anyone talk about it in a while.

(Or not, happy to keep it w/the updated link.)

**Review by negz — APPROVED** [review]
LGTM, modulo the ongoing thread with @hasheddan getting resolved.

I'd prefer to squash/remove all the merge commits to cleanup the commits in this PR before merging.

File: docs/contributing/docs.md
Comment (plumbis) [reply docs/contributing/docs.md]: Development section rewritten to cover make and local builds. https://github.com/crossplane/crossplane/pull/3486/files#diff-93426d953f03b733adbbb574ffd2717d0892773a464c0d301d1b6c9095d67b15R58

File: docs/faqs/related_projects.md
Comment (plumbis) [reply docs/faqs/related_projects.md:78]: I'll keep it for this PR and remove it in a future update.

**Review by hasheddan — APPROVED** [review]
We will squash and merge here per @plumbis request 👍🏻

**Pull request merged**
